libcontainer/nsenter/nsexec.c

```diff
@@ -77,12 +78,18 @@ static int child_func(void *_arg)
     longjmp(*arg->env, 1);
 }
 
-static int clone_parent(jmp_buf *env, int flags) __attribute__((noinline));
-static int clone_parent(jmp_buf *env, int flags)
+static int clone_parent(jmp_buf *env, int flags, bool delay_ipc_unshare) __attribute__((noinline));
+static int clone_parent(jmp_buf * env, int flags, bool delay_ipc_unshare)
 {
 	struct clone_arg ca;
 	int		 child;
 
+	// Don't clone into NEWIPC at the same time as cloning into NEWUSER.
+	// This way we can ensure that NEWIPC namespace belongs to the root in new user namespace.
+	if (delay_ipc_unshare) {
```

Comment:
Yup. It turns out that #960 also can be fixed with some code from my rootless containers patchset too. I'm also cleaning up the `netlink` code to be easier to read.